### PR TITLE
Add new TeamsEventNotificationV2 supporting Microsoft Workflows

### PR DESCRIPTION
**Pull request by kingzacko1**
## Description

Microsoft will be [retiring O365 Connectors on October 1st](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/) in favor of Workflows. Since the current Microsoft Teams Notification relies on connectors, they too will stop working. This PR creates a new `TeamsEventNotificationV2` that supports feeding an AdaptiveCard to the `Post to a channel when a webhook request is received` workflow. Also deprecates `TeamsEventNotification` classes since they will be made effectively useless. 

Though we usually do not backport new feature work, this new feature is effectively a bugfix for a breaking change outside of our control so it will need to be backported to 6.0 and 5.2 in the September bugfix release to allow users time to switch their notifications over before the October 1st shutoff date.
 
## Motivation and Context

see linked article.
closes Graylog2/graylog-plugin-enterprise#7794

## How Has This Been Tested?

local dev environment.
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by ryan-carroll-graylog** [comment]
@kingzacko1 nice job on this! This all looks solid and I'm able to get notifications fireing but for some reason I'm getting this message in teams instead of the card rendering: 
image

Any idea why? I'm just using the default Graylog notification and defaults for [Post to a channel when a webhook request is received](https://make.preview.powerautomate.com/galleries/public/templates/d271a6f01c2545a28348d8f2cddf4c8f/post-to-a-channel-when-a-webhook-request-is-received).

**Comment by kingzacko1** [comment]
@ryan-carroll-graylog could be any number of reasons. I fought with Teams trying to build the silly workflow properly for longer than I care to write publicly. Did you happen to use Post as User or Post as Flow bot? I could only get User to work. I don't know if our teams isn't configured for Flow bot or what, but any time I used Flow bot it failed.

**Comment by ryan-carroll-graylog** [comment]
> @ryan-carroll-graylog could be any number of reasons. I fought with Teams trying to build the silly workflow properly for longer than I care to write publicly. Did you happen to use Post as User or Post as Flow bot? I could only get User to work. I don't know if our teams isn't configured for Flow bot or what, but any time I used Flow bot it failed.

Right on, yeah it is about as far from straightforward as things get.. Looks like it was configured to post as flow bot so I'll give it a shot with user.

**Review by ryan-carroll-graylog — COMMENTED** [review]
This looks great and tests successfully for me @kingzacko1! Thanks for the info up on the Microsoft side setup.

Left one comment about the replay link, but other than that everything looks solid within my testing abilities.

File: graylog2-server/src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationV2.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @VisibleForTesting
+    Map<String, Object> getCustomMessageModel(EventNotificationContext ctx, String type, List<MessageSummary> backlog, DateTimeZone timeZone) {
+        final EventNotificationModelData modelData = EventNotificationModelData.of(ctx, backlog);
+        LOG.debug("Custom message model: {}", modelData);
+
+        final Map<String, Object> objectMap = objectMapperProvider.getForTimeZone(timeZone).convertValue(modelData, TypeReferences.MAP_STRING_OBJECT);
+        objectMap.put("type", type);
+        objectMap.put("http_external_uri", this.httpExternalUri);
```
[the remark this answers is not in the document]
Comment (kingzacko1) [reply graylog2-server/src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationV2.java:148]: I noticed the same thing, but if you check out the [EmailEventNotificationConfig](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/events/notifications/types/EmailEventNotificationConfig.java#L57) template this is actually correct for non-local-dev installs.

[the remark this answers is not in the document]
Comment (ryan-carroll-graylog) [reply graylog2-server/src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationV2.java:148]: Ah nice ok, pretty cool feature I've never noticed before anywhere lol.

**Review by ryan-carroll-graylog — APPROVED** [review]
All LGTM!

**Review by danotorrey — APPROVED** [review]

**Pull request merged**